lib/BackgroundJob/RetentionJob.php

```diff
@@ -197,8 +199,21 @@ private function checkFileId(int $fileId): Node {
 			throw new NotFoundException("No mount points found for file $fileId");
 		}
 
-		$mountPoint = array_shift($mountPoints);
+		foreach ($mountPoints as $mountPoint) {
+			try {
+				return $this->getDeletableNodeFromMountPoint($mountPoint, $fileId);
+			} catch (NotPermittedException $e) {
+				// Check the next mount point
+				$this->logger->debug('Mount point ' . $mountPoint->getMountId() . ' has no delete permissions for file ' . $fileId);
```

Comment:
`ICachedMountInfo::getMountId()` (inherited by `ICachedMountFileInfo`) [can be null](https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/lib/public/Files/Config/ICachedMountInfo.php#L67), so it would be better to use [`ICachedMountInfo::getMountPoint()`](https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/lib/public/Files/Config/ICachedMountInfo.php#L59) instead (or in addition to) to always get a value.

Reply:
but getMountPoint is a path unique to the user and therefor not as helpful iirc.
I will instead add a fallback for null

Reply:
From what I could see `getMountPoint()` includes the user name and the path in that user's filesystem, so that information would help in case debugging is needed, so I would print that instead of `null` :-)